features/netsocket/emac_stack_mem.h

```diff
@@ -29,88 +31,81 @@
  */
 typedef void emac_stack_mem_t;
 typedef void emac_stack_mem_chain_t;
-typedef void emac_stack_t;
 
 /**
  * Allocates stack memory
  *
- * @param  stack Emac stack context
  * @param  size  Size of memory to allocate
  * @param  align Memory alignment requirements
  * @return       Allocated memory struct, or NULL in case of error
  */
-emac_stack_mem_t *emac_stack_mem_alloc(emac_stack_t* stack, uint32_t size, uint32_t align);
+emac_stack_mem_t *emac_stack_mem_alloc(uint32_t size, uint32_t align);
```

Comment:
Should there be an option to have an allocator per network stack? If not, this API will force all network stacks to use lwip's pbuf mechanism (or copy the data).

Reply:
Yes, allocator absolutely needs to be per-stack, or this won't take off. The current mem API will need a little generalisation work anyway, so we've yet to tackle it.

That's in contrast to the other APIs which remain largely unchanged apart from being objectified and C++ified.

I guess we stick with the assumption that each interface is only being used by 1 stack at once, so the interface knows how to pre-allocate its RX buffers.

(In the other system I did this stuff we did permit multiple stacks (IP/NetBIOS/Appletalk/...) to use an interface simultaneously, but everyone was forced to use the same central allocator.)